ios/engine/KMEI/KeymanEngine/Classes/InputViewController.swift

```diff
@@ -351,7 +351,14 @@ open class InputViewController: UIInputViewController, KeymanWebDelegate {
       perform(#selector(self.enableInputClickSound), with: nil, afterDelay: 0.1)
     }
 
-    if numCharsToDelete <= 0 {
+    if numCharsToRightDelete > 0 {
+      for _ in 0..<numCharsToRightDelete {
+        textDocumentProxy.adjustTextPosition(byCharacterOffset: 1)
+        textDocumentProxy.deleteBackward()
+      }
+    }
+
```

Comment:
This should _probably_ be done more properly.  It was sufficient for initial testing, at least.

Unfortunately, there is no `deleteForward` command on that object, hence the text-position shenanigans.

Reply:
It's not possible to `adjustTextPosition(byCharacterOffset: numCharsToRightDelete)`?

I am guessing this needs some real testing for interactions with clusters, SMP because it's likely that the cursor cannot be positioned in the middle of a cluster.

Reply:
I'm sure it's possible to do that... but it may make any SMP-double-checking even rougher.

I'll definitely need to reference this block:

https://github.com/keymanapp/keyman/blob/58a2bda5344a4fa153f772614d644f30b919ec46/ios/engine/KMEI/KeymanEngine/Classes/InputViewController.swift#L366-L378

Noting the step-by-step deletion procedure it follows... and the need to compare/contrast [Swift's preferred UTF-8 representation](https://swift.org/blog/utf8-string/) and UTF-16, it's probably best to go step-by-step here too.